consensus/propagation/types/types_fuzz_test.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import (
+	"testing"
+
+	fuzz "github.com/google/gofuzz"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	"github.com/tendermint/tendermint/libs/rand"
+	"github.com/tendermint/tendermint/proto/tendermint/crypto"
+	protobits "github.com/tendermint/tendermint/proto/tendermint/libs/bits"
+	"github.com/tendermint/tendermint/proto/tendermint/propagation"
+	prototypes "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// TestFuzzMsgFromProto tests the MsgFromProto function with expected message types
+// and mutates them with high probability of nil values
+func TestFuzzMsgFromProto(t *testing.T) {
```

Comment:
how long does this test take? do we need to run this every run? 

imo, if we know of an edge case and its easy enough to include in the correctness test, then we should add that